Engine/Graphics/DecalBuilder.h

```diff
@@ -38,130 +34,101 @@ struct Bloodsplat {
     unsigned char r;
     unsigned char g;
     unsigned char b;
-    char field_1B;
-    int field_1C;  // flags
-    unsigned long long field_20;  // fade time
-    // int field_24;
+    int blood_flags;  // &1 for fade
+    uint64_t fade_timer;
 };
 #pragma pack(pop)
 
-/*  159 */
+// store for all the bloodsplats to be applied
 #pragma pack(push, 1)
 struct BloodsplatContainer {
     inline BloodsplatContainer() {
-        /*_eh_vector_constructor_iterator_(
-          this->std__vector_pBloodsplats,
-          40,
-          64,
-          (void ( *)(void *))Bloodsplat::Bloodsplat,
-          (void ( *)(void *))Bloodsplat::dtor);
-        v1->std__vector_pBloodsplats_size = 0;*/
         uNumBloodsplats = 0;
     }
 
-    virtual ~BloodsplatContainer() {
-        // this->vdestructor_ptr = &BloodsplatContainer_pvdtor;
-        // _eh_vector_destructor_iterator_(this->std__vector_pBloodsplats, 40,
-        // 64, Bloodsplat::dtor);
-    }
+    virtual ~BloodsplatContainer() { }
 
     void AddBloodsplat(float x, float y, float z, float r, float g, float b,
-                       float radius, int a8, int a9);
+        float radius, int a8, int a9);
     void AddBloodsplat(float x, float y, float z, float radius, unsigned char r,
-                       unsigned char g, unsigned char b);
+        unsigned char g, unsigned char b);
 
-    // void ( ***vdestructor_ptr)(BloodsplatContainer *, bool);
-    Bloodsplat std__vector_pBloodsplats[64];
-    unsigned int std__vector_pBloodsplats_size = 0;
-    int uNumBloodsplats;
+    Bloodsplat pBloodsplats_to_apply[64];
+    uint uNumBloodsplats;  // this loops round so old bloodsplats are replaced
 };
 #pragma pack(pop)
 
-#pragma pack(push, 1)
-struct DecalBuilder_stru0 {
```

Comment:
so you merged that with Decal, or was it the same thing all along?

Reply:
Same structure but wasnt used anywhere.